src/lib.mo

```diff
@@ -181,6 +181,33 @@ module {
     var i_element = vec.i_element;
   };
 
+  /// Creates and returns a new vector, populated with the results of calling a provided function on every element in the provided vector
+  ///
+  /// Example:
+  /// ```motoko
+  ///
+  /// vec.add(1);
+  ///
+  /// let t = Vector.map<Nat, Text>(vec, Nat.toText);
+  /// Vector.toArray(t); // => ["1"]
+  /// ```
+  ///
+  /// Runtime: `O(size)`
+  public func map<X1, X2>(vec : Vector<X1>, f : (X1) -> X2) : Vector<X2> = {
```

Comment:
I think the parenthesis around X1 are not needed.

Reply:
removed